autoscheduler/frontend/src/components/SchedulingPage/Schedule/Schedule.tsx

```diff
@@ -228,7 +260,8 @@ const Schedule: React.FC = () => {
           time1,
           time2,
         }));
-        dispatch(mergeThenSelectAvailability({
+        dispatch(mergeAvailability());
+        dispatch(selectMergedAvailability({
           dayOfWeek: startDay,
           available: availabilityMode,
           time1,
```

Comment:
What does this achieve? I deleted it thinking it might have something to do with the same start/end time bug (it didn't lol) but it didn't seem to change any of the functionality, and it still passed all the tests. Also this is the only time `selectMergedAvailability` is called, so I'm curious why it's necessary

Reply:
The `selectMergedAvailability` action exists to deal with a very specific edge case. When updating selected availabilities, the correct av to update is identified by the `time1` fed into the action, as this is where the drag originated and should be unchanged from the original av that is currently in the Redux. As the selected av is continuously dragged, `time2` is always changing, but its `time1` is always the same, and can therefore be used as a unique identifier. 

Unless. Unless the av is merged immediately after being created, in which case, its `time1` might actually become the start/end time of the av it was merged with. Now when would such a thing occur? Pretty much the only situation I can think of is the one that you mentioned earlier, where you click outside of an av but pick a start time where another av already exists. In that case, a new av is created shortly, then immediately merged with the other one so that you actually end up dragging that one.

This oculd be resolved by applying either of the suggested changes of identifying availabilities by ID rather than time, and/or making drag handles size themselves dynamically. In either case, we could theoretically remove this funky `selectMergedAvailability` action.

As for why this isn't included in tests, I remember trying to test this but giving up earlier. I would be willing to give it another shot if y'all think it's worth.

Reply:
Given that it's a very specific edge case, then I think it would be worth adding tests for (assuming it doesn't take like an hour to add them lol) since a "very specific edge case" is probably something that's like to be missed when we're QA testing anytime we change this going forward

Reply:
Btw, turns out the bug was actually fixed by lines 240-253 and this function appears to be unnecessary. Removing the ridiculously clunky `selectMergedAvailability` action

Reply:
Also, I realized that we do have a test for the bug that I was thinking of, and it's on line 336 of `Availability.test.tsx`